tests/util/translator/helper.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    EElementType,
+    IColumns,
+    IConditionalRendering,
+    IPrefillAttributes,
+    IValidation,
+    TElement,
+    TElementMap,
+} from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data";
+import { ISchemaConditionalRendering } from "src/translator";
+import { TextBasedField } from "src/translator/text-based-field";
+
+interface MockElementProps {
+    type: EElementType;
+    id: string;
+    internalId: string;
+    required?: boolean;
+    requiredErrorMsg?: string;
+    columns?: IColumns;
+    placeholder?: string;
+    label?: string;
+    validation?: IValidation[];
+    conditionalRendering?: IConditionalRendering[];
+    prefill?: IPrefillAttributes[];
+}
+
+interface MockElementSchemaProps {
+    id: string;
+    label: string;
+    uiType: EElementType;
+    placeholder?: string;
+    validation?: TextBasedField.ISchemaValidation[];
+    showIf?: ISchemaConditionalRendering[];
+}
+
+interface MockPrefillSchema {
+    [key: string]: IPrefillAttributes[];
+}
+
+interface MockChildrenSchema {
+    [key: string]: {
+        label: string;
+        uiType: EElementType;
+        columns: IColumns;
+        placeholder?: string;
+        validation?: TextBasedField.ISchemaValidation[];
+        showIf?: ISchemaConditionalRendering[];
+    };
+}
+
+export const generateMockElement = ({
+    type,
+    id,
+    internalId,
+    required,
+    requiredErrorMsg,
+    columns,
+    placeholder,
+    label,
+    validation,
+    conditionalRendering,
+    prefill,
+}: MockElementProps): TElementMap => {
+    return {
+        [id]: {
+            internalId,
+            type,
+            id,
+            required: required ?? true,
+            requiredErrorMsg: requiredErrorMsg || "",
+            columns: columns || { desktop: 12, tablet: 8, mobile: 4 },
+            placeholder: placeholder || "",
+            label: label || ELEMENT_BUTTON_LABELS[type],
+            validation: validation || [],
+            conditionalRendering: conditionalRendering || [],
+            prefill: prefill || [],
+        },
+    };
+};
```

Comment:
this can be simplified

```suggestion
export const generateMockElement = (props: MockElementProps): TElementMap => {
    return {
        [props.id]: {
            required: true,
            columns: { desktop: 12, tablet: 8, mobile: 4 },
            label: ELEMENT_BUTTON_LABELS[props.type],
            ...props,
        },
    };
};
```